src/bin/zmtrainsa

```diff
@@ -117,6 +117,13 @@ fi
 source `dirname $0`/zmshutil || exit 1
 zmsetvars
 
+export TMPDIR=${zmtrainsa_tmp_directory:-${zimbra_tmp_directory}/zmtrainsa}
```

Comment:
No problem, I change it accordingly. I'll then change the mktemp calls as well so that the generated directories are prefixed with zmtrainsa, ok?

My rationale for a subdirectory was that if something goes wrong and somebody wants to clean up the left over temp directories, they won't accidently delete other stuff as well (been there done that). A prefix would have the same result though.

I actually never noticed that `/opt/zimbra/data/tmp` is `1777`. I consider this a design defect for exactly the same reasons you gace: There are a lot of hard coded paths in that drectory which are partly also generated on demand. And it it isn't really required (all users who write there should either be `zimbra` or `root`). That's a different issue though.